cmd/soroban-cli/src/commands/contract/bump.rs

```diff
@@ -144,10 +151,30 @@ impl Cmd {
             tracing::debug!(?events);
         }
 
-        Ok(())
+        // The transaction from core will succeed regardless of whether it actually found & bumped
+        // the entry, so we have to inspect the result meta to tell if it worked or not.
+        let TransactionMeta::V3(TransactionMetaV3 { operations, .. }) = meta else {
+            return Err(Error::LedgerEntryNotFound);
+        };
+
+        // Simply check if there is exactly one entry here. We only support bumping a single
+        // entry via this command (which we should fix separately, but).
+        if operations.len() == 0 {
+            return Err(Error::LedgerEntryNotFound);
+        }
+
+        if operations[0].changes.len() != 2 {
+            return Err(Error::LedgerEntryNotFound);
+        }
+
+        let (LedgerEntryChange::State(_state), LedgerEntryChange::Updated(LedgerEntry{ data: LedgerEntryData::ContractData(ContractDataEntry{expiration_ledger_seq, ..}), ..})) = (&operations[0].changes[0], &operations[0].changes[1]) else {
```

Comment:
what in the abomination is this un `cargo fmt`d line

Reply:
It actually *is* cargo fmt'ed